nbs/04_processing.ipynb

```diff
@@ -203,61 +205,74 @@
+    "                focus_area=self.focus_area,\n",
+    "                grid_configs=self.grid_configs)\n",
+    "            label_id_overview_fig.savefig(self.results_dir_path.joinpath('ROI_label_IDs_overview.png'), dpi=300)\n",
     "            plt.close()\n",
     "        if config.save_summary_results:\n",
     "            self._create_and_save_csv_result_files(activity_filtered_analysis_rois)\n",
     "            self._create_and_save_individual_traces_pdf_result_file(activity_filtered_analysis_rois)\n",
     "        if config.save_single_trace_results:\n",
     "            self._create_and_save_single_trace_results_as_csv(activity_filtered_analysis_rois)\n",
     "        if config.export_to_nwb:\n",
-    "            self._export_all_data_to_nwb()\n",
+    "            self._export_all_data_to_nwb(nwb_metadata)\n",
     " \n",
-    "    def _export_all_data_to_nwb(self) -> None:\n",
-    "        nwb_file = self._initialize_nwb_file()\n",
-    "        nwb_file = self._add_source_data(nwb_file)\n",
+    "    def _export_all_data_to_nwb(self, nwb_metadata: dict[str, Any] = None) -> None:\n",
+    "        if nwb_metadata is None:\n",
+    "            nwb_metadata = {}\n",
+    "        nwb_file = self._initialize_nwb_file(nwb_metadata)\n",
+    "        nwb_file = self._add_source_data(nwb_file, nwb_metadata)\n",
     "        nwb_file = self._add_na3_results(nwb_file)\n",
     "        self._write_nwb_file(nwb_file)\n",
     "\n",
     "\n",
-    "    def _initialize_nwb_file(self) -> NWBFile:\n",
+    "    def _initialize_nwb_file(self, nwb_metadata: dict[str, Any]) -> NWBFile:\n",
     "        nwb_file = NWBFile(\n",
-    "            session_description=\"Autogenerated NWB file by neuralactivitycubic\",\n",
+    "            session_description=\"Autogenerated NWB file by Neuralactivitycubic\",\n",
     "            identifier=str(uuid4()),\n",
-    "            session_start_time=datetime.now(tzlocal())\n",
+    "            session_start_time=datetime.now(tzlocal()),\n",
+    "            experimenter=nwb_metadata.get('experimenter', 'NA3 user'),\n",
+    "            experiment_description=nwb_metadata.get('experiment_description', 'Experiment done with NA3'),\n",
+    "            lab=nwb_metadata.get('lab', 'NA3 lab'),\n",
+    "            institution=nwb_metadata.get('institution', 'NA3 institution'),\n",
+    "            notes=nwb_metadata.get('notes', ''),\n",
     "        )\n",
     "        return nwb_file\n",
     "\n",
     "\n",
-    "    def _add_source_data(self, nwb_file: NWBFile) -> NWBFile:\n",
+    "    def _add_source_data(self, nwb_file: NWBFile, nwb_metadata: dict[str, Any]) -> NWBFile:\n",
     "        # Add recording source data\n",
-    "        microscope = nwb_file.create_device(name = 'Microscope')\n",
+    "        microscope = nwb_file.create_device(\n",
+    "            name = 'Microscope',\n",
+    "            model_number=nwb_metadata.get('model_number', 'NA3 microscope'),\n",
```

Comment:
Same as above, leave blank if no information was provided or add check if field has to be omitted.